### ISSUE TITLE
Bug in the Base64 encoding

### ISSUE THREAD

**Issue by hrmrules** [kubernetes-sigs/kustomize#2191]
There seems to be a bug in the Base64 encoding used to add secrets and configmaps with the appropriate generator.
It seems the problem is related to the character 0x0a in the file. We encountered this problem when trying to add binary keystore files to secrets.

The following is a minimal example to illustrate the problem.

Version: Kustomize v3.5.4
Sample kustomization.yaml:
```
configMapGenerator:
  - name: TestB64ConfigMap
    files:
      - binfile
```
Contents of binfile: 
`echo SIsNCgE= |  base64 -d > binfile `
where SIsNCgE= is the proper base64 value

Sample output of `kustomize build`
```
apiVersion: v1
binaryData:
  binfile: SIsKAQ==
kind: ConfigMap
metadata:
  name: TestB64ConfigMap-bfg8g4dc27
```

The base64 value which should have been SIsNCgE= is now SIsKAQ==
The base64 decoding happens according to the proper rules, so the wrong base64 encoding yields a corrupted binary in the pod.

**Comment by no2a** [comment]
configmap/secret generator break binary files (I believe it is a bug). see #2125

**Comment by hefekranz** [comment]
same issue, using
{Version:kustomize/v3.5.4 GitCommit:3af514fa9f85430f0c1557c4a0291e62112ab026 BuildDate:2020-01-11T03:12:59Z GoOs:linux GoArch:amd64}

suddenly our generated secrets from binary data were broken

**Comment by ppihus** [comment]
Same here.
`{Version:3.5.4 GitCommit:3af514fa9f85430f0c1557c4a0291e62112ab026 BuildDate:2020-01-11T06:57:29+00:00 GoOs:darwin GoArch:amd64}`

**Comment by fotex** [comment]
Same issue on version
{Version:3.5.4 GitCommit:3af514fa9f85430f0c1557c4a0291e62112ab026 BuildDate:2020-01-17T14:28:58+00:00 GoOs:darwin GoArch:amd64}

**Comment by vanbalken** [comment]
Same here with version 3.5.4. I've reverted back to version 3.5.3 to get it working again.

**Comment by Shell32-Natsu** [comment]
I cannot reproduce this issue after v3.5.5. Please verify whether it has been fixed in 3.5.5.

**Comment by Shell32-Natsu** [comment]
Please reopen if it's not fixed.

/close
